torch_geometric/utils/subgraph.py

```diff
@@ -170,6 +170,7 @@ def bipartite_subgraph(
     if src_subset.dtype != torch.bool:
         src_size = int(edge_index[0].max()) + 1 if size is None else size[0]
         src_subset = index_to_mask(src_subset, size=src_size)
+    if dst_subset.dtype != torch.bool:
```

Comment:
Lets add a test where `src_subset` is of type bool and `dst_subset` is of type int.

Reply:
Sounds good. I just added the test case. I also expanded the test of `HeteroData.subgraph` to cover this issue.